src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_TypeInference.cs

```diff
@@ -0,0 +1,1341 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    public partial class StaticNullChecking : CSharpTestBase
```

Comment:
If you don't have tests for array type inference already, could you add some? `var t = new [] { null, string.Empty };` and variations. 
Also include an example with non-ref type, to capture current behavior (`var t = new [ ] { 1, null }; // I expect error`).

Reply:
Added.